Gems/Atom/Feature/Common/Code/Source/Mesh/MeshFeatureProcessor.cpp

```diff
@@ -163,11 +165,77 @@ namespace AZ
         void MeshFeatureProcessor::OnBeginPrepareRender()
         {
             m_meshDataChecker.soft_lock();
+
+            // Must read cvar from AZ::Console due to static variable in multiple libraries, see ghi-5537
+            bool enablePerMeshShaderOptionFlagsCvar = false;
+            if (auto* console = AZ::Interface<AZ::IConsole>::Get(); console != nullptr)
+            {
+                console->GetCvarValue("r_enablePerMeshShaderOptionFlags", enablePerMeshShaderOptionFlagsCvar);
+            }
```

Comment:
If it's not okay to check r_enablePerMeshShaderOptionFlags directly here, then should each of the light feature processors take this same approach as well?

Reply:
yeah good catch. This just affects things when the cvar is set using a setreg, but still important.

Reply:
I don't really understand what you mean. What does IConsole have to do with setreg? Maybe this indicates we need a better code comment to explain what's going on.